client/src/App.js

```diff
@@ -1,134 +1,74 @@
-        console.log(data);
-      } catch (error) {
-        console.error(error);
-      }
-    };
-
-    fetchData();
-  }, []);
-  return (
-    <React.Fragment>
-      <Sidebar />
-      <Header
-        searchQuery={searchQuery}
-        handleChanges={handleChanges}
-        handleSubmit={handleSubmit}
-      />
-      <Home {...props} />
-      <Footer />
-    </React.Fragment>
-  );
-}
-
-function VideoListPage(props) {
-  return (
-    <React.Fragment>
-      <Sidebar />
-      <Header />
-      <VideoList {...props} />
-      <Footer />
-    </React.Fragment>
-  );
-}
-
-function VideoPlayerPage(props) {
-  return (
-    <React.Fragment>
-      <Sidebar />
-      <Header />
-      <Comments />
-      <VideoPlayer {...props} />
-      <Footer />
-    </React.Fragment>
-  );
-}
-
-
-function EditProfilePage(props) {
-  return (
-    <React.Fragment>
-      <Sidebar />
-      <Header />
-      <EditProfile {...props} />
-      <Footer />
-    </React.Fragment>
-  );
-}
-
-function ProfilePage(props) {
-  return (
-    <React.Fragment>
-      <Sidebar />
-      <Header />
-      <Profile {...props} />
-      <Footer />
-    </React.Fragment>
-  );
-}
+import history from "./history";
 
 function App() {
+  const [uploadedVideos, setUploadedVideos] = useState([]);
+  
   return (
     <ApolloProvider client={client}>
       <Router history={history}>
-      <Switch>
-    <Route path="/home" component={HomePage} />
-    <Route path="/videolist" component={VideoListPage} />
-    <Route path="/videoplayer" component={VideoPlayerPage} />
-    <Route path="/editprofile" component={EditProfilePage} />
-    <Route path="/profile" component={ProfilePage} />
-    <Route path="/signup" component={SignUp} />
-    <Route path="/login" component={Login} />
-  
-    
-</Switch>
-
+        <Switch>
+          <Route path="/home">
+            <Sidebar />
+            <Header />
+            <Home
+              uploadedVideos={uploadedVideos}
+              setUploadedVideos={setUploadedVideos}
+            />
+            <Footer />
+          </Route>
+          <Route path="/videolist">
+            <Sidebar />
+            <Header />
+            <VideoList />
+            <Footer />
+          </Route>
+          <Route path="/videoplayer">
+            <Sidebar />
+            <Header />
+            <Comments />
+            <VideoPlayer />
+            <Footer />
+          </Route>
+          <Route path="/editprofile">
+            <Sidebar />
+            <Header />
+            <EditProfile />
+            <Footer />
+          </Route>
+          {/* <Route path="/logout">
+            <Header />
+            <Sidebar />
+            <Logout />
+            <Footer />
+          </Route> */}
+          <Route path="/profile">
+            <Sidebar />
+            <Header />
+            <Profile />
+            <Footer />
+          </Route>
+          <Route path="/">
+            <Login />
+          </Route>
+        </Switch>
```

Comment:
This code we would add to each page to get header, footer, sidebar. Lines 117-127 should be correct code for implementation